pkg/datapath/linux/config/config.go

```diff
@@ -140,6 +132,48 @@ func (h *HeaderfileWriter) WriteClusterConfig(w io.Writer) error {
 	return fw.Flush()
 }
 
+// HeaderfileReader is a wrapper type which implements datapath.ConfigReader.
+// It manages reading and validation of datapath program headerfiles.
+type HeaderfileReader struct{}
+
+func NewHeaderfileReader() *HeaderfileReader {
+	return &HeaderfileReader{}
+}
+
+// ValdateClusterConfig reads an existing cluster-wide options config file and
+// validates it against the given configuration.
+func (h *HeaderfileReader) ValidateClusterConfig(r io.Reader, cfg datapath.ClusterConfiguration) error {
+	cDefinesMap := cfg.GetOptions()
+
+	for key, currentVal := range h.ReadConfigMacros(r) {
+		if newVal, ok := cDefinesMap[key]; ok && newVal != currentVal {
+			return fmt.Errorf("cluster_config.h cannot be changed. %s was defined as %s, but new agent configuration would change it to %v", key, currentVal, newVal)
+		}
+	}
```

Comment:
IIUC, this is validating the cluster config file that comes from clustermesh, each time a new cluster is added to the mesh, right? If that's the case, adding that context somewhere is helpful in tying the high-level of clustermesh down to these details of the code to make it easier for developers to understand the full picture of the code.

Reply:
This part is actually checking an existing `cluster_config.h` header that's on disk, and making sure its values haven't changed. This is to prevent a user from breaking their cluster if they try to enable `max-connected-clusters=511` on an existing cluster thats running on the default 255. I'll update the comments here to make it more clear